e2e/e2etests/test_migrate_tss.go

```diff
@@ -0,0 +1,182 @@
+package e2etests
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"time"
+
+	sdkmath "cosmossdk.io/math"
+	"github.com/btcsuite/btcutil"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/e2e/runner"
+	"github.com/zeta-chain/zetacore/e2e/utils"
+	zetacrypto "github.com/zeta-chain/zetacore/pkg/crypto"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func TestMigrateTss(r *runner.E2ERunner, _ []string) {
+	r.SetBtcAddress(r.Name, false)
+	stop := r.MineBlocksIfLocalBitcoin()
+	defer stop()
+
+	// Pause inbound procoessing for tss migration
+	r.Logger.Info("Pause inbound  processing")
+	msg := observertypes.NewMsgDisableCCTX(
+		r.ZetaTxServer.GetAccountAddress(0),
+		false,
+		true)
+	_, err := r.ZetaTxServer.BroadcastTx(utils.FungibleAdminName, msg)
+	require.NoError(r, err)
+
+	// Migrate btc
+	// Fetch balance of BTC TSS address
+	utxos, err := r.GetTop20UTXOsForTssAddress()
+	require.NoError(r, err)
+
+	var btcBalance float64
+	for _, utxo := range utxos {
+		btcBalance += utxo.Amount
+	}
+
+	// Use fixed fee for migration
+	btcTSSBalanceOld := btcBalance
+	fees := 0.01
+	btcBalance -= fees
+	btcChain := int64(18444)
+
+	//migrate btc funds
+	// #nosec G701 e2eTest - always in range
+	migrationAmountBTC := sdkmath.NewUint(uint64(btcBalance * 1e8))
+	msgMigrateFunds := crosschaintypes.NewMsgMigrateTssFunds(
+		r.ZetaTxServer.GetAccountAddress(0),
+		btcChain,
+		migrationAmountBTC,
+	)
+	_, err = r.ZetaTxServer.BroadcastTx(utils.FungibleAdminName, msgMigrateFunds)
+	require.NoError(r, err)
+
+	// Fetch migrator cctx for btc migration
+	migrator, err := r.ObserverClient.TssFundsMigratorInfo(r.Ctx, &observertypes.QueryTssFundsMigratorInfoRequest{
+		ChainId: btcChain})
+	require.NoError(r, err)
+	cctxBTCMigration := migrator.TssFundsMigrator.MigrationCctxIndex
+
+	// ETH migration
+	// Fetch balance of ETH TSS address
+	tssBalance, err := r.EVMClient.BalanceAt(context.Background(), r.TSSAddress, nil)
+	require.NoError(r, err)
+	ethTSSBalanceOld := tssBalance
+
+	tssBalanceUint := sdkmath.NewUintFromString(tssBalance.String())
+	evmChainID, err := r.EVMClient.ChainID(context.Background())
```

Comment:
We can reuse the runner context.
We should ideally almost never user Background